tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/AppServiceTests.cs

```diff
@@ -0,0 +1,157 @@
+            string resource, 
+            string userAssignedClientIdOrResourceId, 
+            UserAssignedIdentityId userAssignedIdentityId = UserAssignedIdentityId.None)
+        {
+            Environment.SetEnvironmentVariable("IDENTITY_ENDPOINT", endpoint);
+            Environment.SetEnvironmentVariable("IDENTITY_HEADER", "secret");
+
+            using (var httpManager = new MockHttpManager())
+            {
+                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
+                    .Create("clientId")
+                    .WithHttpManager(httpManager)
+                    .WithExperimentalFeatures()
+                    .Build();
+
+                httpManager.AddManagedIdentityMockHandler(
+                    endpoint, 
+                    resource, 
+                    MockHelpers.GetMsiSuccessfulResponse(), 
+                    userAssignedClientIdOrResourceId: userAssignedClientIdOrResourceId, 
+                    userAssignedIdentityId: userAssignedIdentityId);
+
+                var result = await cca.AcquireTokenForClient(new string[] { scope })
+                    .WithManagedIdentity(userAssignedClientIdOrResourceId)
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                Assert.IsNotNull(result);
+                Assert.IsNotNull(result.AccessToken);
+                Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+                result = await cca.AcquireTokenForClient(new string[] { scope })
+                    .WithManagedIdentity()
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                Assert.IsNotNull(result);
+                Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow("user.read")]
+        [DataRow("https://management.core.windows.net//user_impersonation")]
+        [DataRow("s")]
+        public async Task AppServiceTestWrongScopeAsync(string resource)
+        {
+            Environment.SetEnvironmentVariable("IDENTITY_ENDPOINT", "http://127.0.0.1:41564/msi/token");
+            Environment.SetEnvironmentVariable("IDENTITY_HEADER", "secret");
+
+            using (var httpManager = new MockHttpManager())
+            {
+                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
+                    .Create("clientId")
+                    .WithHttpManager(httpManager)
+                    .WithExperimentalFeatures()
+                    .Build();
+
+                httpManager.AddManagedIdentityMockHandler("http://127.0.0.1:41564/msi/token", resource, MockHelpers.GetMsiErrorResponse(), statusCode: HttpStatusCode.InternalServerError);
+                httpManager.AddManagedIdentityMockHandler("http://127.0.0.1:41564/msi/token", resource, MockHelpers.GetMsiErrorResponse(), statusCode: HttpStatusCode.InternalServerError);
```

Comment:
Can we have a test / modify this test to get a 500 error without a payload and assert the exception message?